tests/unit/runner/test_service.py

```diff
@@ -119,3 +124,106 @@ def test_proxy_dapp_service(test_utils, service_kwargs, expected_attrs, expected
             assert getattr(proxy_service, name) == value
     except Exception as e:
         test_utils.verify_error(expected_exc, e)
+
+
+SOME_PAYLOAD: Final = Payload()
+SOME_NETWORK: Final = Network(Mock(), "192.168.0.1/24", "192.168.0.1")
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "descriptor, payloads, networks, base_class, expected_kwargs, error",
+    (
+        # non-networked service
+        (
+            ServiceDescriptor(entrypoint=[], payload="foo"),
+            {"foo": SOME_PAYLOAD},
+            {},
+            DappService,
+            {"payload": SOME_PAYLOAD, "instance_params": [{"entrypoint": []}]},
+            None,
+        ),
+        # networked service
+        (
+            ServiceDescriptor(entrypoint=[], payload="foo", network="bar"),
+            {"foo": SOME_PAYLOAD},
+            {"bar": SOME_NETWORK},
+            DappService,
+            {
+                "instance_params": [{"entrypoint": []}],
+                "payload": SOME_PAYLOAD,
+                "network": SOME_NETWORK,
+            },
+            None,
+        ),
+        # missing payload definition
+        (
+            ServiceDescriptor(entrypoint=[], payload="foo"),
+            {},
+            {},
+            None,
+            {},
+            RunnerError('Undefined payload: "foo"'),
+        ),
+        # missing network definition
+        (
+            ServiceDescriptor(entrypoint=[], payload="foo", network="bar"),
+            {"foo": SOME_PAYLOAD},
+            {},
+            None,
+            {},
+            RunnerError('Undefined network: "bar"'),
+        ),
+        # network service with an http proxy
+        (
+            ServiceDescriptor(
+                entrypoint=[],
+                payload="foo",
+                network="bar",
+                http_proxy=HttpProxyDescriptor(ports=[PortMapping(80)]),
+            ),
+            {"foo": SOME_PAYLOAD},
+            {"bar": SOME_NETWORK},
+            HttpProxyDappService,
+            {
+                "instance_params": [{"entrypoint": [], "remote_port": 80}],
+                "payload": SOME_PAYLOAD,
+                "network": SOME_NETWORK,
+            },
+            None,
+        ),
+        # explicit IP address
+        (
+            ServiceDescriptor(
+                entrypoint=[],
+                payload="foo",
+                network="bar",
+                ip=["192.168.0.2"],
+            ),
+            {"foo": SOME_PAYLOAD},
+            {"bar": SOME_NETWORK},
+            DappService,
+            {
+                "instance_params": [{"entrypoint": []}],
+                "payload": SOME_PAYLOAD,
+                "network": SOME_NETWORK,
+                "network_addresses": ["192.168.0.2"],
+            },
+            None,
+        ),
+    ),
+)
```

Comment:
I think a scenario with:
* empty `networks`
* `http-proxy`

could be useful here, but I'm not sure - maybe this is already tested somewhere.

Reply:
yes, it is tested in `test_http_proxy_descriptor`